### PR TITLE
ONEUP-1870: split theme service into details/summaries

### PR DESCRIPTION
**Pull request by Normal**

### CODE REVIEW

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    when(mockRequest.execute(anyString())).thenReturn(Future {
+      responseMock
+    })
+
+    when(metricsMock.defaultRegistry).thenReturn(new com.codahale.metrics.MetricRegistry())
+    when(mockWsClient.url(anyString)).thenReturn(mockRequest)
+  }
+
+  "ThemeDetailsService" should {
```
Comment (catalin-cretu) [inline pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala:50]: I don't know if we forgot to refactor this name or the IDE didn't pick it up, but should this be `DetailsThemeService` instead?

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    val responseMock: WSResponse = mock[WSResponse]
+    val metricsMock: Metrics = mock[Metrics]
+    val mockTimerContext: Context = mock[Context]
+
+    when(mockRequest.withHttpHeaders(ArgumentMatchers.any())).thenReturn(mockRequest)
+    when(mockRequest.addHttpHeaders(ArgumentMatchers.any())).thenReturn(mockRequest)
+    when(mockRequest.withQueryStringParameters(ArgumentMatchers.any())).thenReturn(mockRequest)
+    when(mockRequest.withAuth(anyString, anyString, ArgumentMatchers.eq(WSAuthScheme.BASIC))).thenReturn(mockRequest)
+    when(mockRequest.withBody(anyString)(ArgumentMatchers.any())).thenReturn(mockRequest)
+    when(mockRequest.withRequestTimeout(ArgumentMatchers.any())).thenReturn(mockRequest)
```
Comment (catalin-cretu) [inline pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala:40]: Have you looked into using [Mockito#RETURNS_SELF](https://javadoc.io/static/org.mockito/mockito-core/3.5.7/org/mockito/Mockito.html#RETURNS_SELF) for mocking `AhcWSRequest`'s fluent API?

**Review by catalin-cretu — APPROVED** [review]

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala (hunk 1 above)
Comment (Normal) [reply pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala:50]: yes, you are right

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    when(mockWsClient.url(anyString)).thenReturn(mockRequest)
+  }
+
+  "ThemeDetailsService" should {
+
+    "provide content and related data by theme path" in new TestScope {
+      val service = new DetailsThemeServiceIml(mockWsClient, metricsMock, executionContext)
+
+      when(responseMock.status).thenReturn(Status.OK)
+      when(responseMock.json).thenReturn(JsObject(
```
Comment (aramin) [inline pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala:49]: your version is fine but json could also be created by using the `Json.obj` method. it's a little less verbose.
 e.g.
```
val json: JsValue = Json.obj(
  "name" -> "Watership Down",
  "location" -> Json.obj("lat" -> 51.235685, "long" -> -1.309197),
  "residents" -> Json.arr(
    Json.obj(
      "name" -> "Fiver",
      "age" -> 4,
      "role" -> JsNull
    ),
    Json.obj(
      "name" -> "Bigwig",
      "age" -> 6,
      "role" -> "Owsla"
    )
  )
)
```

Comment (Normal) [reply pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala:49]: yeah this looks much better, will do

**Review by aramin — APPROVED** [review]

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala (hunk 3 above)
Comment (Normal) [reply pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala]: fixed

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala (hunk 2 above)
Comment (Normal) [reply pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala]: didn't know about it, fixed

File: pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala (hunk 1 above)
Comment (Normal) [reply pressed/src/test/scala/de/welt/contentapi/pressed/client/services/DetailsThemeServiceTest.scala]: fixed

**Pull request merged**